### PR TITLE
Java 21 and Springboot 3 Migration

### PR DESCRIPTION
**Pull request by hwatkins-ch**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/security/WebSecurityConfigurer.java [hunk 1]
```diff
@@ -15,28 +16,30 @@ public class WebSecurityConfigurer {
...
-    public static class GovUkAccountsSecurityFilterConfig extends WebSecurityConfigurerAdapter {
+    public static class GovUkAccountsSecurityFilterConfig {
 
-        @Override
-        protected void configure(HttpSecurity http)
-                throws Exception {
+        @Bean
+        public SecurityFilterChain govUkAccountsSecurityFilterChain(HttpSecurity http) throws Exception {
+            http
+                .securityMatcher("/accounts/**")
```
Comment (dhawes-ch) [inline src/main/java/uk/gov/companieshouse/web/accounts/security/WebSecurityConfigurer.java:24]: Think this should be a requestMatcher to match previous behaviour, unless the intention is to exclusively apply security to /accounts/* endpoints for some reason

Comment (hwatkins-ch) [reply src/main/java/uk/gov/companieshouse/web/accounts/security/WebSecurityConfigurer.java:24]: Good spot, will do

Comment (hwatkins-ch) [reply src/main/java/uk/gov/companieshouse/web/accounts/security/WebSecurityConfigurer.java]: Done

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/DirectorsReportAdditionalInformationControllerTest.java [hunk 2]
```diff
@@ -20,14 +20,13 @@
...
-import javax.servlet.http.HttpSession;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpSession;
 import java.util.List;
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/DirectorsReportAdditionalInformationControllerTest.java:29]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/FinancialCommitmentsControllerTest.java [hunk 3]
```diff
@@ -20,14 +20,13 @@
...
-import javax.servlet.http.HttpSession;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpSession;
 import java.util.List;
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/FinancialCommitmentsControllerTest.java:29]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java [hunk 4]
```diff
@@ -27,9 +27,7 @@
 
 import java.util.List;
 
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyBoolean;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAmortisationPolicyControllerTests.java:30]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAssetsNoteControllerTest.java [hunk 5]
```diff
@@ -2,8 +2,8 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/IntangibleAssetsNoteControllerTest.java:5]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/LoansToDirectorsAdditionalInfoControllerTest.java [hunk 6]
```diff
@@ -2,8 +2,8 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/LoansToDirectorsAdditionalInfoControllerTest.java:5]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/OffBalanceSheetArrangementsControllerTest.java [hunk 7]
```diff
@@ -2,8 +2,7 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/OffBalanceSheetArrangementsControllerTest.java:5]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/OtherAccountingPolicyControllerTest.java [hunk 8]
```diff
@@ -28,9 +28,7 @@
 
 import java.util.List;
 
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyBoolean;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/OtherAccountingPolicyControllerTest.java:31]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/PoliticalAndCharitableDonationsControllerTest.java [hunk 9]
```diff
@@ -20,14 +20,14 @@
...
-import javax.servlet.http.HttpSession;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpSession;
 import java.util.List;
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/PoliticalAndCharitableDonationsControllerTest.java:29]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/PrincipalActivitiesControllerTest.java [hunk 10]
```diff
@@ -2,8 +2,7 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/PrincipalActivitiesControllerTest.java:5]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/PrincipalActivitiesSelectionControllerTest.java [hunk 11]
```diff
@@ -2,8 +2,7 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/PrincipalActivitiesSelectionControllerTest.java:5]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/ProfitAndLossControllerTests.java [hunk 12]
```diff
@@ -2,8 +2,8 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/ProfitAndLossControllerTests.java:5]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/TangibleDepreciationPolicyControllerTest.java [hunk 13]
```diff
@@ -28,9 +28,7 @@
 
 import java.util.List;
 
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyBoolean;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/TangibleDepreciationPolicyControllerTest.java:31]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/TurnoverPolicyControllerTest.java [hunk 14]
```diff
@@ -28,20 +29,11 @@
...
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyBoolean;
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.doThrow;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/TurnoverPolicyControllerTest.java:32]: wildcard

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/ValuationInformationPolicyControllerTest.java [hunk 15]
```diff
@@ -28,9 +28,7 @@
 
 import java.util.List;
 
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyBoolean;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/ValuationInformationPolicyControllerTest.java:31]: wildcard

File: pom.xml [hunk 16]
```diff
@@ -306,15 +312,22 @@
...
+                <version>${jib-maven-plugin.version}</version>
                 <configuration>
                     <container>
                         <expandClasspathDependencies>true</expandClasspathDependencies>
                     </container>
+                    <from>
+                        <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/ci-corretto-build-21:latest</image>
+                    </from>
+                    <to>
+                        <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/company-accounts.web.ch.gov.uk:latest</image>
```
[the remark this answers is not in the document]
Comment (hwatkins-ch) [reply pom.xml:329]: Hmmmmmm good question

**Review by akunati-ch — APPROVED** [review]

File: pom.xml (hunk 16 above)
[the remark this answers is not in the document]
Comment (hwatkins-ch) [reply pom.xml]: Uusing old image reference until staging deployment done

**Review by dgurjar-ch — APPROVED** [review]

**Pull request merged**